### PR TITLE
A single function to get CompositionTimeOffset for both RTMP and eRTMP

### PR DESCRIPTION
**Pull request by harryz2000**
This will fix the hevc pts issue.

### CODE REVIEW

**Comment by bcostdolby** [comment]
Maybe remove the Get/Set AVCTS methods and replace with GetCTS and SetCTS that are not AVC specific but based on if it is extended or not will set/get the correct byte of the extra data. 

Right now we are saying it is extended and then modifying the meaning of extra data.

File: src/rtmp/rtmpmessage.cpp [hunk 1]
```diff
@@ -978,6 +978,21 @@ void RTMPVideoFrame::Dump()
 	}
 }
 
+int32_t RTMPVideoFrame::GetCompositionTime() const
+{
+	if (isExtended)
+	{
+		if (GetVideoCodecEx() == VideoCodecEx::HEVC || GetVideoCodecEx() == VideoCodecEx::H264)
+			return  GetSI24(extraData);
```
Comment (bcostdolby) [inline src/rtmp/rtmpmessage.cpp:986]: Is this correct as a SI24? I assume that means signed? The original CTS was a unsigned

Comment (harryz2000) [reply src/rtmp/rtmpmessage.cpp:986]: it is signed yes.

SI24                           ¦Signed 24-bit integer

Comment (harryz2000) [reply src/rtmp/rtmpmessage.cpp:986]: ¦      // See ISO/IEC 14496-12, 8.15.3 for an explanation of composition times.                                                                                           ¦
¦      // The offset in an FLV file is always in milliseconds.                                                                                                            ¦
¦      compositionTimeOffset = SI24                                                                                                                                       ¦
¦

**Comment by harryz2000** [comment]
> Maybe remove the Get/Set AVCTS methods and replace with GetCTS and SetCTS that are not AVC specific but based on if it is extended or not will set/get the correct byte of the extra data.
> 
> Right now we are saying it is extended and then modifying the meaning of extra data.

Added @todo. We should remove the AVC functions. yes.

**Comment by harryz2000** [comment]
> Maybe remove the Get/Set AVCTS methods and replace with GetCTS and SetCTS that are not AVC specific but based on if it is extended or not will set/get the correct byte of the extra data.
> 
> Right now we are saying it is extended and then modifying the meaning of extra data.

Added todo to remove AVC specific functions. Renamed function to GetComponsitionTime().

**Review by bcostdolby — APPROVED** [review]
lgtm

**Review by murillo128 — APPROVED** [review]
lgtm

**Pull request merged**